rts/motoko-rts/src/region.rs

```diff
@@ -315,58 +310,44 @@ mod meta_data {
 
         use super::{bytes_of, offset, size};
         use crate::region::{BlockId, RegionId};
-        use crate::stable_mem::{read_u16, write_u16};
+        use crate::stable_mem::{read_u16, read_u64, read_u8, write_u16, write_u64, write_u8};
 
         // Compute an offset in stable memory for a particular block ID (based zero).
         fn index(b: &BlockId) -> u64 {
             offset::BLOCK_REGION_TABLE + (b.0 as u64 * size::BLOCK_REGION_TABLE_ENTRY as u64)
         }
 
-        /// Some(r, j) means that the block is in use by region r, at slot j.
+        /// Some(r, j, c) means that the block is in use by region r, at slot j with c allocated pages.
         /// None means that the block is available for (re-)allocation.
-        pub fn get(b: BlockId) -> Option<(RegionId, u16)> {
-            let offset = index(&b);
-            let raw = read_u16(offset);
-            let rid = RegionId::from_u16(raw);
-            rid.map(|r| (r, read_u16(offset + bytes_of::<u16>())))
+        pub fn get(b: BlockId) -> Option<(RegionId, u16, u8)> {
+            let region_offset = index(&b);
+            let rank_offset = region_offset + bytes_of::<u64>();
+            let page_count_offset = rank_offset + bytes_of::<u16>();
+            let region = read_u64(region_offset);
+            let rid = RegionId::from_u64(region);
+            rid.map(|r| (r, read_u16(rank_offset), read_u8(page_count_offset)))
         }
 
-        /// Some(r, j) means that the block is in use by region r, at slot j.
+        /// Some(r, j, c) means that the block is in use by region r, at slot j with c allocated pages.
         /// None means that the block is available for (re-)allocation.
-        pub fn set(b: BlockId, r: Option<(RegionId, u16)>) {
+        pub fn set(b: BlockId, r: Option<(RegionId, u16, u8)>) {
+            let region_offset = index(&b);
+            let rank_offset = region_offset + bytes_of::<u64>();
+            let page_count_offset = rank_offset + bytes_of::<u16>();
             match r {
-                None => write_u16(index(&b), RegionId::into_u16(None)),
-                Some((r, j)) => {
-                    let offset = index(&b);
-                    write_u16(offset, RegionId::into_u16(Some(r)));
-                    write_u16(offset + bytes_of::<u16>(), j)
+                None => {
+                    write_u64(region_offset, RegionId::into_u64(None));
+                    write_u16(rank_offset, 0);
+                    write_u8(page_count_offset, 0);
+                }
+                Some((r, j, c)) => {
+                    write_u64(region_offset, RegionId::into_u64(Some(r)));
+                    write_u16(rank_offset, j);
+                    write_u8(page_count_offset, c);
                 }
             }
```

Comment:
Just cosmetics: Maybe, the write duplicates could be reduced:
(sorry, code suggestion does not work here as it spans code line additions and deletions)
```
            let (region_id, rank, allocated_pages) = match r {
                None => (None, 0, 0),
                Some((r, j, c)) => (Some(r), j, c),
            };
            write_u64(region_offset, RegionId::into_u64(region_id));
            write_u16(rank_offset, rank);
            write_u8(page_count_offset, allocated_pages);
```

Reply:
I'll do that